### PR TITLE
Pendingbids

### PR DESCRIPTION
**Pull request by alexvandesande**
* Checks pending bids to see if they are already saved on the bids collections, and either they should be
* Changes average to median
* Shows last chance to bid only if less than 24 hours
* Only shows finalize button to owners
* Adds a fading yellow to new items on the list
* Prevent database from getting too big

### CODE REVIEW

File: app/imports/lib/ethereum.js [hunk 1]
```diff
@@ -335,6 +335,36 @@ export default ethereum = (function() {
           }})        
       })
 
+      var lastDay = Math.floor(new Date().getTime()) - 24 * 60 * 60 + 10 * 60 * 1000;
```
Comment (Arachnid) [inline app/imports/lib/ethereum.js]: I'm fairly sure you meant `(24 * 60 * 60 + 10 * 60) * 1000`.

File: app/imports/lib/ethereum.js [hunk 2]
```diff
@@ -335,6 +335,36 @@ export default ethereum = (function() {
...
+      _.each(PendingBids.find({date: {$gt: lastDay}}).fetch(), ( bid, i) => {  
+        // check for duplicates 
+        var dupBid = MyBids.find({shaBid:bid.shaBid}).fetch();
+        if (dupBid && dupBid.shaBid == bid.shaBid && dupBid.secret == bid.secret){
+            console.log('removing duplicate bid for', bid.name)
+            PendingBids.remove({_id: bid._id});            
+        } else {
+          registrar.contract.sealedBids.call(bid.owner, bid.shaBid, (err, result) => {
+            if (err) {
+              console.log('Error looking bid', bid.name, err);
```
Comment (Arachnid) [inline app/imports/lib/ethereum.js]: s/looking/looking for/

File: app/imports/templates/components/nameStatus.js [hunk 3]
```diff
@@ -161,19 +161,21 @@ Template['components_nameStatus'].helpers({
...
       return Names.find({registrationDate: {$gt: revealDeadline}, name:{$gt: ''}},{}).count() > 0;
     },
-    averageValue() {
-      var average = _.reduce(
-          Names.find({value: {$gt:0.01}}).fetch(), function(memo,num) { 
-            return memo + num.value; 
-          }, 0);
-      return Math.round(average*100/Names.find({value: {$gt:0.01}}).count())/100 || '--';
+    medianValue() {
+      var disputedNames = Names.find({value: {$gt:0.01}}, {sort: {value: 1}}).fetch();
```
Comment (Arachnid) [inline app/imports/templates/components/nameStatus.js:165]: I'm worried this will scale poorly; how big can `disputedNames` get?

Comment (alexvandesande) [reply app/imports/templates/components/nameStatus.js:165]: I'm adding an unrelated new bit where it limits the amount of registered names it keeps an eye on to 50.

File: app/imports/templates/components/nameStatus.js [hunk 4]
```diff
@@ -144,36 +144,38 @@ Template['components_nameStatus'].helpers({
...
     }, 
     percentageDisputed() {
       return Math.round(100 - (100 * Names.find({value: {$gt:0.01}}).count() / Names.find({value: {$gt:0}}).count())) || 0;
     },
     canBeInvalidated(name) {
       return name.length < 7;
     },
+    recent(registrationDate) {
+      var diff = Math.floor(new Date().getTime()/1000) - registrationDate;      
+      return ((diff < 600 && diff > 0)|| (diff < 60-5*24*60*60)) ? 'recent' : '';
```
Comment (Arachnid) [inline app/imports/templates/components/nameStatus.js]: Doesn't the first condition imply the second? `60-5*24*60*60` is negative.

Comment (Arachnid) [reply app/imports/templates/components/nameStatus.js]: Never mind, I see how the second condition could be true when the first isn't; how is that 'recent', though?

Comment (alexvandesande) [reply app/imports/templates/components/nameStatus.js]: The first condition shows recently registered, the second shows names about to end their bid phase

**Review by Arachnid — APPROVED** [review]

**Pull request merged**